app/code/core/Mage/Adminhtml/Block/Sales/Order/Create/Search/Grid.php

```diff
@@ -104,13 +104,14 @@ protected function _prepareCollection()
             ->setStore($this->getStore())
             ->addAttributeToSelect($attributes)
             ->addAttributeToSelect('sku')
+            ->addAttributeToSelect('gift_message_available')
             ->addStoreFilter()
             ->addAttributeToFilter('type_id', array_keys(
                 Mage::getConfig()->getNode('adminhtml/sales/order/create/available_product_types')->asArray()
             ))
-            ->addAttributeToSelect('gift_message_available');
-
-        Mage::getSingleton('catalog/product_status')->addSaleableFilterToCollection($collection);
+            ->addAttributeToFilter('status', [
+                'in' => Mage::getModel('catalog/product_status')->getSaleableStatusIds()
```

Comment:
Creating a new instance of `catalog/product_status` every time is not relevant here, so we can keep `getSingleton`. Same in other places.

Reply:
perfectly right, just done it